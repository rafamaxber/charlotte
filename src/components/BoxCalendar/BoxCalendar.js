import React from 'react';
import styled from 'styled-components';
import moment from 'moment';
import Button from '../Button/Button';
import BoxTitle from '../BoxTitle/BoxTitle';

const Box = styled.div`
  width: 100%;
  max-width: 838px;
  margin: auto;
  background: #FFFFFF;
  box-shadow: 0px -40px 46px rgba(0, 0, 0, 0.332484);
  position: relative;
  padding-right: 20px;
  padding-left: 20px;
  @media (min-width: 570px) {
    border-radius: 7px;
    padding-right: 60px;
    padding-left: 60px;
  }
`;

const BoxDate = styled.div`
  @media (max-width: 570px) {
    width: 50%;
    border: 1px solid rgb(206, 255, 142);
    margin-bottom: 30px;
    padding: 10px 0;
  }
`;
const Text = styled.div`
  font-family: 'Heebo', sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: normal;
  letter-spacing: 2.5px;
`
const BoxDateTitle = Text.extend`
  color: #555555;
  font-size: 16px;
  text-tranform: uppercase;
  @media (min-width: 570px) {
    font-size: 22px;
  }
`;

const BoxDateValue = Text.extend`
  font-size: 15px;
  letter-spacing: 1.36364px;
  color: #B5B5B5;
  @media (min-width: 570px) {
    font-size: 20px;
  }
`;
const WrapperBoxDate = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  text-align: center;
  align-items: center;
  justify-content: center;
  @media (min-width: 570px) {
    flex-direction: column;
    justify-content: space-evenly;
  }
`;
const WrapperBoxCalendar = styled.div`
  position: relative;
  z-index: 0;
`;
const WrapperBox = styled.div`
  display: flex;
  justify-content: space-around;
  margin-top: 50px;
  @media (max-width: 570px) {
    margin-top: 30px;
    flex-wrap: wrap-reverse;
  }
`;

const dateTranslate = (dateIso) => {
  if (dateIso) {
    const utcDate = dateIso.toUTCString();
    return (
      <span>
        {moment(utcDate).format("MMMM")}
        <b> {moment(utcDate).format("DD")}</b>,
        <span> {moment(utcDate).format("YYYY")} </span>
      </span>
    );
  }
  return false;
}


export default ({ children, startDate, endDate, fetchHotels}) => (
  <Box>
    <BoxTitle>
      Select the dates to stay in Charlotte
    </BoxTitle>

    <WrapperBox>
      <WrapperBoxDate>
        <BoxDate>
          <BoxDateTitle>
            CHECK-IN
          </BoxDateTitle>
          <BoxDateValue>
            { dateTranslate(startDate) || `Choose a date` } 
          </BoxDateValue>
        </BoxDate>

        <BoxDate>
          <BoxDateTitle>
            CHECK-OUT
          </BoxDateTitle>
          <BoxDateValue>
            {dateTranslate(endDate) || `Choose a date`} 
          </BoxDateValue>
        </BoxDate>

        <Button typeBtn="default" size="default" action={fetchHotels}>
          Search hotels
        </Button>

      </WrapperBoxDate>
      
      <WrapperBoxCalendar>
        {children}
      </WrapperBoxCalendar>
    
    </WrapperBox>

  </Box>
);
