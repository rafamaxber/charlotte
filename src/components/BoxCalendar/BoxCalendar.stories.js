import React from 'react';
import { storiesOf } from '@storybook/react';
import BoxCalendar from './BoxCalendar';

storiesOf('BoxCalendar', module)
  .add('BoxCalendar', () => (
    <BoxCalendar></BoxCalendar>
  ));
