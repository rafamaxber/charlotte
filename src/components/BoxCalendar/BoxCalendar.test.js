import React from 'react';
import renderer from 'react-test-renderer';
import BoxCalendar from './BoxCalendar';

describe('BoxCalendar', () => {
  it('Should match with snapshot', () => {
    const component = renderer.create(
      <BoxCalendar></BoxCalendar>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
