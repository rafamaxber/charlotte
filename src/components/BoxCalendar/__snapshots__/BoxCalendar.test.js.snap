// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxCalendar Should match with snapshot 1`] = `
<div
  className="sc-ifAKCX cMizZd"
>
  <div
    className="sc-htpNat cPXLHW"
  >
    <div
      className="sc-bxivhb dxsIBs"
    >
      Select the dates to stay in Charlotte
    </div>
  </div>
  <div
    className="sc-gZMcBi eSihRx"
  >
    <div
      className="sc-dnqmqq brpsMC"
    >
      <div
        className="sc-EHOje edRFCv"
      >
        <div
          className="sc-gzVnrw iOgjDl"
        >
          CHECK-IN
        </div>
        <div
          className="sc-htoDjs gmNAtx"
        >
          Choose a date
        </div>
      </div>
      <div
        className="sc-EHOje edRFCv"
      >
        <div
          className="sc-gzVnrw iOgjDl"
        >
          CHECK-OUT
        </div>
        <div
          className="sc-htoDjs gmNAtx"
        >
          Choose a date
        </div>
      </div>
      <a
        className="sc-bdVaJa fxfHPe"
        onClick={undefined}
      >
        Search hotels
      </a>
    </div>
    <div
      className="sc-iwsKbI hcoZJr"
    />
  </div>
</div>
`;
